platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1507,7 +1507,23 @@ - (NSExpression *)mgl_expressionLocalizedIntoLocale:(nullable NSLocale *)locale
                         localizedKeyPath = [NSString stringWithFormat:@"name_%@", preferredLanguage];
                     }
                 }
-                return [NSExpression expressionForKeyPath:localizedKeyPath];
+                // If the keypath is `name` or `name_en`, no need to fallback
+                if ([localizedKeyPath isEqualToString:@"name"] || [localizedKeyPath isEqualToString:@"name_en"]) {
+                    return [NSExpression expressionForKeyPath:localizedKeyPath];
+                }
+                // If the keypath is `name_zh-Hans`, fallback to `name_zh`. If `name_zh` is empty, fallback to `name`
+                // The `name_zh-Hans` field was added since Mapbox Streets v7
+                // See the documentation of name fields for detail https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview
+                if ([localizedKeyPath isEqualToString:@"name_zh-Hans"]) {
```

Comment:
Apparently the same `mapbox.mapbox-streets-v7` tileset on api.mapbox.cn uses `name_zh-CN` instead of `name_zh-Hans`: mapbox/mapbox-navigation-ios#1554. Could we address that here by opportunistically falling back to `name_zh-CN` if `name_zh-Hans` is `nil`?

/cc @m-stephen